inference-engine/src/multi_device/multi_device_plugin.cpp

```diff
@@ -142,13 +142,26 @@ InferenceEngine::Parameter MultiDeviceInferencePlugin::GetMetric(const std::stri
     }
 }
 
+// Is called only when caching is enabled
+    InferenceEngine::IExecutableNetworkInternal::Ptr MultiDeviceInferencePlugin::LoadNetwork(const std::string& modelPath,
+                                                                           const std::map<std::string, std::string>& config) {
+    CNNNetwork network;
+    return LoadExeNetworkImpl(modelPath, network, config);
+}
+
 ExecutableNetworkInternal::Ptr MultiDeviceInferencePlugin::LoadExeNetworkImpl(const CNNNetwork &network,
                                                                               const std::map<std::string, std::string>& config) {
+    return LoadExeNetworkImpl({}, network, config);
+}
+
+ExecutableNetworkInternal::Ptr MultiDeviceInferencePlugin::LoadExeNetworkImpl(const std::string& modelPath,
+                                                                              CNNNetwork network,
```

Comment:
whta is the reason to pass the network in (esp by value)?

Reply:
Copy is just a new shared_ptr to same internal implementation. Copy of CNNNetwork object is widely used across the project

Regarding "by value" - it simplifies code a bit, please look on line 217, in case if network is not loaded and we have to load it - in this case change this 'network' with loaded CNNNetwork. Using 'const-reference' will require to declare 'CNNNetwork loadedNetwork' object, copy it to 'network' and set it on line #217